lib/src/eol.rs

```diff
@@ -40,28 +117,157 @@ fn is_binary(bytes: &[u8]) -> bool {
     false
 }
 
+enum ConvertMode {
+    /// Apply EOL conversion on both snapshot and update. Equivalent to
+    /// `eol=crlf`.
+    InputOutput,
+    /// Apply EOL conversion only on snapshot. Equivalent to `eol=lf`.
+    Input,
+}
+
+enum EffectiveEolMode {
```

Comment:
nit: add some toplevel doc comment to both enums

Reply:
Done.